Add recipe for incus-tramp

lckarssen opened the pull request:
### Brief summary of what the package does

With this package one can connect to [Incus](https://linuxcontainers.org/incus/) containers using TRAMP. This is basically a copy of [`lxd-tramp`](https://github.com/onixie/lxd-tramp) (packaged [here](https://github.com/melpa/melpa/blob/master/recipes/lxd-tramp)) with calls to `lxc`/`lxd`  replaced with calls to `incus`. 

### Direct link to the package repository

https://gitlab.com/lckarssen/incus-tramp

### Your association with the package

I am the maintainer. 

### Relevant communications with the upstream package maintainer

None needed

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

riscy commented:
Thanks for this.  Quick first pass -

`incus-tramp.el` with byte-compile using Emacs 29.4:
```
incus-tramp.el:81:57: Warning: Unused lexical argument `ignored'
In end of data:
incus-tramp.el:79:19: Warning: the function `cl-evenp' might not be defined at runtime.
```
Instead of `ignored` the convention is to write e.g. `_ignored` or `_` or some other name prefixed with `_`.  You probably want to explicitly depend on `cl-lib` to solve the second.

`incus-tramp.el` with package-lint 20240913.1546:
```
1 issue found:
101:1: warning: `eval-after-load' is for use in configurations, and should rarely be used in packages.
```
This looks (imo) like something best left to the user to do in their config, via some function you provide for them.  It'll help with debuggability of their config - for instance, what if there is a bug in your code that prevents the user from running tramp (I'm not saying there is)? What if there is a bug in their overall tramp packages, and they want to exclude your package from the list of culprits? Autoloading in this modification means that even if the user doesn't `require` your package, their Emacs runtime will still be affected by it.

lckarssen commented:
Thanks for your comments @riscy! Much appreciated. This is my first Emacs package and, although I have been using Emacs for ~20 years, I am not well versed in Lisp. As mentioned in the PR description I basically copied the `lxd-tramp.el` code and did some search-replace work. 

Regarding your remark to make the package depend on `cl-lib`: I had actually made the following change from the original `lxd-tramp` from which I "forked" this package:
```diff
-;; Package-Requires: ((emacs "24.4") (cl-lib "0.6"))
+;; Package-Requires: ((emacs)) 
```
The reason for this change was that with the `cl-lib` dependency `package-lint-this-buffer` gives me the following warning message:
```
9:52: warning: An explicit dependency on cl-lib <= 1.0 is not needed on Emacs >= 24.3.
```

And thanks for the advice about the `eval-after-load` call. I'll try to find a better way.

lckarssen commented:
I have rewritten part of the package code and now it doesn't depend on `cl-lib` any more. This includes removal of the function with the `ignored` argument. 

The `eval-after-load` call has also been removed from the package code and added instructions to the `README.org` so users can configure/load the package themselves. 

See [v1.1](https://gitlab.com/lckarssen/incus-tramp/-/releases/v1.1) for the latest release. This should address all of the concerns @riscy raised above.

riscy commented:
> 9:52: warning: An explicit dependency on cl-lib <= 1.0 is not needed on Emacs >= 24.3.

I think you've sorted it out, but this warning was because of your `Package-Requires` line, but the byte-compile warnings I was pointing out above were due to your `(require ...)` lines.  The former are dependencies that the package needs to have installed (cl-lib is already distributed with Emacs 24+), but the latter are used by the byte-compiler.

This looks good to merge, but I notice the function `incus-tramp--list-running-containers` is both private and not referred to anywhere else in your code - fyi.
